docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+                "status": "greeter_not_found"
+            },
+            {
+                // The greeter is not part of the allowed greeters for this invitation
+                "status": "greeter_not_allowed"
+            }
+        ]
+    }
+]
+```
+
+
+#### `invite_greeter_cancel_attempt` command
+
+Two new commands are introduced to cancel an attempt, which registers a cancellation reason for the given attempt. For the greeter, an authenticated command is exposed. After using `invite_greeter_cancel_attempt`, the greeter can start a new attempt by using the `invite_greeter_start_attempt` command.
+
+In order for a greeter to cancel an attempt, a couple of checks have to be performed:
+- the invitation exists
+- the invitation is pending (i.e. not completed or cancelled)
+- the author is allowed (i.e. it is part of the greeters)
+- the attempt exists
+- the attempt is active (i.e. joined and not cancelled)
+
+Then, the active attempt is cancelled with the provided reason and a new active attempt is created. The cancellation reason covers several cases:
+- the user manually cancelled the attempt
+- the hashed nonce didn't match the provided nonce
+- the SAS code communicated to the user was invalid
+- the payload could not be deciphered
+- the payload could not be deserialized
+- the payload contained invalid information
+- the attempt has been automatically cancelled by a new `start_attempt` command
+
+It is also registered that the greeter cancelled the attempt, so that this information can be provided in subsequent `invitation_cancelled` replies, along with the corresponding reason and timestamp. This way, the front-end application has all the information to properly communicate to the user what happened during the attempt.
+
+Pseudo-code:
+
+```python
+def invite_greeter_cancel_attempt(
+    token: InvitationToken,
+    attempt: AttemptID,
+    reason: CancelledAttemptReason,
+) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  GreeterNotAllowed | OK:
+    # Perform checks
+    [...]
+    # Call `greeter_cancel_attempt(reason)`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_greeter_cancel_attempt",
+            "fields": [
+                {
+                    "name": "token",
+                    "type": "InvitationToken"
+                },
+                {
+                    "name": "attempt",
+                    "type": "AttemptID"
+                },
+                {
+                    "name": "reason",
+                    "type": "CancelledAttemptReason"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok",
+                "fields": [
+                    {
+                        "name": "attempt",
```

Comment:
The attemp ID has been provided in the request, so there is no need to return it !

Reply:
d36e8a3440bf5a5cb645e168de63108650bc36bf